skycatalogs/catalog_creator.py

```diff
@@ -285,26 +424,77 @@ def create_galaxy_catalog(self):
                 return
         self.write_config()
 
+    def _write_subpixel(self, dat=None, output_path=None, arrow_schema=None,
+                        to_rename=dict(), stride=100000):
+        '''
+        Write out data for a single healpixel, single source type
+        Parameters
+        ----------
+        dat           dict    The data to write out. Column names are the keys;
+                              values are numpy array
+        output_path   string  path to output file
+        arrow_schema          Schema for output parquet file
+        stride        int     number of rows to include in a row group
+        to_rename     dict    Associate input column name with output name
+                              if they differ
+        '''
+        dlen = 0
+        for val in dat.values():
+            dlen = len(val)
+            break
+        if dlen == 0: return
+
+        last_row_ix = dlen - 1
+        u_bnd = min(stride, dlen)
+        l_bnd = 0
+        rg_written = 0
+        writer = None
+
+        while u_bnd > l_bnd:
+            out_dict = {k : dat[k][l_bnd : u_bnd] for k in dat if k not in to_rename}
+            for k in to_rename:
+                out_dict[to_rename[k]] = dat[k][l_bnd : u_bnd]
+            out_df = pd.DataFrame.from_dict(out_dict)
+            out_table = pa.Table.from_pandas(out_df, schema=arrow_schema)
+            if not writer:
+                writer = pq.ParquetWriter(output_path, arrow_schema)
+
+            writer.write_table(out_table)
+            rg_written += 1
+            l_bnd = u_bnd
+            u_bnd = min(l_bnd + stride, last_row_ix + 1)
+
+        writer.close()
+        self._logger.debug(f'# row groups written to {output_path}: {rg_written}')
+
     def create_galaxy_pixel(self, pixel, gal_cat, arrow_schema):
         """
         Parameters
         ----------
-        pixel           Pixel for which catalog is to be generated.
+        pixel           Pixel for which catalog(s) is(are) to be generated.
+                        Note: input pixels are assumed to use nside=32. Output
+                        pixels may be finer
         gal_cat         GCRCatalogs-loaded galaxy truth (e.g. cosmoDC2)
         arrow_schema    schema to use for output file
         """
 
-        output_path = os.path.join(self._output_dir, f'galaxy_{pixel}.parquet')
-        if os.path.exists(output_path):
-            if not self._skip_done:
-                self._logger.info(f'Overwriting file {output_path}')
-            else:
-                self._logger.info(f'Skipping regeneration of {output_path}')
-                return
-
-        # Used to finding tophat parameters from cosmoDC2 column names
-        tophat_bulge_re = r'sed_(?P<start>\d+)_(?P<width>\d+)_bulge'
-        tophat_disk_re = r'sed_(?P<start>\d+)_(?P<width>\d+)_disk'
+        if self._galaxy_nside > 32:
```

Comment:
I think it will be true for diffsky catalogs, but in general, it can't be assumed for all time and all catalogs. However currently, as far as I know, there is no way for skyCatalogs to figure out what nside is in the input catalog. It could and perhaps should be added to the GCRCatalogs config file.